packages/flutter/lib/src/material/icon_button.dart

```diff
@@ -343,18 +343,20 @@ class IconButton extends StatelessWidget {
     );
     final BoxConstraints adjustedConstraints = effectiveVisualDensity.effectiveConstraints(unadjustedConstraints);
 
+    final double size = iconSize ?? iconTheme.size!;
```

Comment:
Since the `iconTheme` variable is to be used only once, maybe you should just use IconTheme.of(context) directly.

```suggestion
    final double size = iconSize ?? IconTheme.of(context).size!;
```